### PR TITLE
Maxtext docker

### PR DESCRIPTION
**Pull request by kocchop**
Adding support for MaxText to the JAX-toolbox

### CODE REVIEW

**Comment by abhinavgoel95** [comment]
@terrykong could you review this PR? It adds MaxText support to Jax-Toolbox.

File: .github/workflows/_sandbox.yaml [hunk 1]
```diff
@@ -1,41 +1,127 @@
-name: "~Sandbox"
+name: ~sandbox
```
Comment (terrykong) [inline .github/workflows/_sandbox.yaml:1]: just leaving a reminder to revert before merging

File: .github/container/Dockerfile.maxtext [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+## Download source and add auxiliary scripts
+###############################################################################
+
+FROM ${BASE_IMAGE} as mealkit
+
+ARG REPO_MAXTEXT
+ARG REF_MAXTEXT
+ARG SRC_PATH_MAXTEXT
+
+RUN <<"EOF" bash -ex
```
Comment (terrykong) [inline .github/container/Dockerfile.maxtext:18]: Is there a particular reason to keep this out of the central manifest.yaml? Putting it here means that your `latest_verified_commit` won't be bumped by our CI so JAX will move ahead and maxtext will stay where it is.

File: .github/container/Dockerfile.maxtext [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# syntax=docker/dockerfile:1-labs
+
+ARG BASE_IMAGE=ghcr.io/nvidia/jax:mealkit
+ARG REPO_MAXTEXT=https://github.com/google/maxtext.git
+ARG REF_MAXTEXT=main
```
Comment (terrykong) [inline .github/container/Dockerfile.maxtext:5]: lines 4-5 & 14-15 aren't needed

**Review by terrykong — CHANGES_REQUESTED** [review]

**Comment by terrykong** [comment]
If you'd like one extra guard before enabling building these images in the CI, you could contribute a nightly workflow like this one: https://github.com/NVIDIA/JAX-Toolbox/blob/main/.github/workflows/nightly-pax-build.yaml, except make it run a no-op. Then in a follow up PR you can fill its contents. This is to work around a limitation of Github Actions where you cannot run a workflow that does not exist on the `default` branch, which in our case is `main`

**Review by yhtang — CHANGES_REQUESTED** [review]
Please add an associated build workflow as @terrykong suggested.

**Comment by kocchop** [comment]
> If you'd like one extra guard before enabling building these images in the CI, you could contribute a nightly workflow like this one: https://github.com/NVIDIA/JAX-Toolbox/blob/main/.github/workflows/nightly-pax-build.yaml, except make it run a no-op. Then in a follow up PR you can fill its contents and test it. This is to work around a limitation of Github Actions where you cannot run a workflow that does not exist on the `default` branch, which in our case is `main`

**Comment by kocchop** [comment]
@terrykong @yhtang I have made the necessary changes. I have added a placeholder file for nightly maxtext build. Could you please review and let me know?

File: .github/workflows/nightly-maxtext-build.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: Nightly MaxText build
+'on': workflow_dispatch
```
Comment (terrykong) [inline .github/workflows/nightly-maxtext-build.yaml:2]: I'm not sure how strict GH actions is, but technically on.workflow_dispatch should be a dict. It doesn't need any args, see the sandbox for an example: https://github.com/NVIDIA/JAX-Toolbox/blob/main/.github/workflows/_sandbox.yaml#L4

**Review by terrykong — CHANGES_REQUESTED** [review]

**Review by terrykong — APPROVED** [review]

**Comment by terrykong** [comment]
LGTM

@yhtang any other comments?

**Review by abhinavgoel95 — DISMISSED** [review]

**Review by yhtang — CHANGES_REQUESTED** [review]
Shall we also build it in the presubmit?

Also, I noticed that the Dockerfile is only for amd64. What's the current status for this on arm?

**Comment by kocchop** [comment]
Hi @yhtang currently there is a tensorflow version mismatch for arm. We may need to create a separate docker file with corresponding wheels. Right now focusing on x86 version to be working. Will work towards the arm build soon

**Comment by kocchop** [comment]
Hi @yhtang , I have added maxtext to the presubmit CI. I have also tested the dockerfile with sandbox and it is working for amd64. Please let me know if you have any other suggestions.

**Review by yhtang — APPROVED** [review]
LGTM. Note that we need to add tests (smoke, functional, perf, etc.) via subsequent PRs.

**Pull request merged**